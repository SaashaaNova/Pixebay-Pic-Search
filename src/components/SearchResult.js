import React from "react";
import PropTypes from "prop-types";

const SearchResult = ({ imgs = [] }) => (
  <div className='search-result-cnt'>
    {imgs.map( picture => (
      <div key={picture.id}>
        <a href={picture.pageURL} target="_blank" rel="noopener noreferrer">
          <img src={picture.webformatURL} alt=""/>
        </a>
      </div>
    ))}
  </div>
);

SearchResult.propTypes = {
    imgs: PropTypes.array
};

export default SearchResult;