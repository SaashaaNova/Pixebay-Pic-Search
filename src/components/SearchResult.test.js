import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import SearchResult from "./SearchResult.js";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const mockPics = [
    {
      id: 1,
      pageURL: "https://github.com/",
      webformatURL: "Github"
    }
  ];

  describe("SearchResult component", () => {
    test("renders", () => {
      const wrapper = shallow(<SearchResult imgs={mockPics} />);
  
      expect(wrapper).toMatchSnapshot();
    });

    test("if empty array is returned", () => {
      const wrapper = shallow(<SearchResult />);
  
      expect(wrapper).toMatchSnapshot();
    });

    test("runs when there is no pics", () => {
      const wrapper = shallow(<SearchResult />);
  
      expect(wrapper.find("li")).toHaveLength(0);
    });

    test("runs when array is empty", () => {
      const wrapper = shallow(<SearchResult imgs={[]} />);
  
      expect(wrapper.find("li")).toHaveLength(0);
    });
});
