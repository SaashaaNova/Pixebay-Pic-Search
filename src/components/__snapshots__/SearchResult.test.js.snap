// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult component if empty array is returned 1`] = `ShallowWrapper {}`;

exports[`SearchResult component renders 1`] = `ShallowWrapper {}`;
